rust/windows-client/src-tauri/src/debug_commands.rs

```diff
@@ -0,0 +1,175 @@
+        handle: Option<file_logger::Handle>,
+    }
+
+    impl Callbacks for CallbackHandler {
+        type Error = std::convert::Infallible;
+
+        fn on_disconnect(&self, error: Option<&Error>) -> Result<(), Self::Error> {
+            tracing::error!("on_disconnect not implemented. Error: {error:?}");
+            Ok(())
+        }
+
+        fn on_error(&self, error: &Error) -> Result<(), Self::Error> {
+            tracing::error!("on_error not implemented. Error: {error}");
+            Ok(())
+        }
+
+        fn on_update_resources(
+            &self,
+            _resource_list: Vec<ResourceDescription>,
+        ) -> Result<(), Self::Error> {
+            tracing::error!("on_update_resources not implemented");
+            Ok(())
+        }
+
+        fn roll_log_file(&self) -> Option<PathBuf> {
+            self.handle
+                .as_ref()?
+                .roll_to_new_file()
+                .unwrap_or_else(|e| {
+                    tracing::debug!("Failed to roll over to new file: {e}");
+                    let _ = self.on_error(&Error::LogFileRollError(e));
+
+                    None
+                })
+        }
+    }
+
+    let (layer, handle) = file_logger::layer(std::path::Path::new("."));
+    setup_global_subscriber(layer);
+
+    let device_id = crate::device_id::get();
+
+    let mut session = Session::connect(
+        common_args.api_url,
+        SecretString::from(common_args.token),
+        device_id,
+        CallbackHandler {
+            handle: Some(handle),
+        },
+    )
+    .unwrap();
+
+    tracing::info!("new_session");
+
+    block_on_ctrl_c();
+
+    session.disconnect(None);
+    Ok(())
+}
+
+/// Test encrypted credential storage
+pub fn credentials() -> Result<()> {
+    use keyring::Entry;
+
+    // TODO: Remove placeholder email
+    let entry = Entry::new_with_target("token", "firezone_windows_client", "username@example.com")?;
+    match entry.get_password() {
+        Ok(password) => {
+            println!("Placeholder password is '{password}'");
+
+            println!("Deleting password");
+            entry.delete_password()?;
+        }
+        Err(keyring::Error::NoEntry) => {
+            println!("No password in credential manager");
+
+            let new_password = "top_secret_password";
+            println!("Setting password to {new_password}");
+            entry.set_password(new_password)?;
+        }
+        Err(e) => return Err(e.into()),
+    }
+
+    Ok(())
+}
+
+/// Test generating a device ID from the BIOS or MAC address. This should survive OS re-installs and uniquely identify a device.
+pub fn device_id() -> Result<()> {
+    use smbioslib::SMBiosSystemInformation as SysInfo;
+
+    let data = smbioslib::table_load_from_device()?;
+    if let Some(uuid) = data.find_map(|sys_info: SysInfo| sys_info.uuid()) {
+        println!("SMBios uuid: {uuid}");
+    } else {
+        println!("SMBios couldn't find uuid");
+    }
+
+    Ok(())
+}
```

Comment:
So I chatted more with @conectado about our approach here and I think I led you astray with the spec on this.

Cases where the SMBios isn't usable or isn't unique:

- Docker containers (not a big issue for end-user client like Windows)
- VMs (SMBios serial is not unique)
- Hypervisors (??)

I think we should just generate a random UUID instead and persist it to disk. See the [`store` plugin](https://github.com/tauri-apps/plugins-workspace/tree/v1/plugins/store#usage-from-rust)